src/mod_caps.erl

```diff
@@ -225,17 +239,18 @@ disco_info(Acc = #{node := Node}) ->
         false -> Acc
     end.
 
-c2s_presence_in(C2SState,
-                {From, To, Packet = #xmlel{attrs = Attrs, children = Els}}) ->
+-spec c2s_presence_in(ejabberd_c2s:state(), jid:jid(), jid:jid(), exml:element()) ->
+          ejabberd_c2s:state().
+c2s_presence_in(C2SState, From, To, Packet = #xmlel{attrs = Attrs, children = Els}) ->
     ?LOG_DEBUG(#{what => caps_c2s_presence_in,
                  to => jid:to_binary(To), from => jid:to_binary(From),
                  exml_packet => Packet, c2s_state => C2SState}),
     Type = xml:get_attr_s(<<"type">>, Attrs),
     Subscription = ejabberd_c2s:get_subscription(From, C2SState),
-    Insert = ((Type == <<"">>) or (Type == <<"available">>))
-        and ((Subscription == both) or (Subscription == to)),
-    Delete = (Type == <<"unavailable">>) or (Type == <<"error">>),
-    case Insert or Delete of
+    Insert = (Type == <<>> orelse Type == <<"available">>)
+        and (Subscription == both orelse Subscription == to),
+    Delete = Type == <<"unavailable">> orelse Type == <<"error">>,
+    case Insert orelse Delete of
```

Comment:
I usually prefer `or`, when operations on booleans are made. because of better type checking at runtime
```
5> true or 4.
** exception error: bad argument
     in operator  or/2
        called as true or 4
6> true orelse 4.
true
```

Reply:
On the other hand, short-circuit expressions are tail-recursive, quicker to execute and they make it obvious to the reader that the side-effects are not important. In this case I could go with both, `orelse` is just more natural to me, there is no complex type-checking involved here as the expressions are simple. I could revert, no problem for me.